routes/error-tracker.js

```diff
@@ -61,95 +99,54 @@ function logEvent(log, event) {
  * @param {!Object<string, string>} params
  * @return {?Promise} May return a promise that rejects on logging error
  */
-function handler(req, res, params) {
+async function handler(req, res) {
   const referrer = req.get('Referrer');
+  const params = getRequestParams(req);
   const reportingParams = extractReportingParams(params);
-  const {
-    debug,
-    message,
-    buildQueryString,
-    stacktrace,
-    version,
-  } = reportingParams;
+  const { debug, message, version } = reportingParams;
   const logTarget = new LogTarget(referrer, reportingParams);
 
+  // Reject requests missing essential info.
   if (!referrer || !version || !message) {
-    res.sendStatus(statusCodes.BAD_REQUEST);
-    return null;
+    return res.sendStatus(statusCodes.BAD_REQUEST);
   }
+  // Accept but ignore requests that get throttled.
   if (
     version.includes('internalRuntimeVersion') ||
     Math.random() > logTarget.throttleRate
   ) {
-    res.sendStatus(statusCodes.OK);
-    return null;
+    return res.sendStatus(statusCodes.OK);
   }
 
-  return latestRtv().then(rtvs => {
-    if (rtvs.length > 0 && !rtvs.includes(version)) {
-      res.sendStatus(statusCodes.OK);
-      return null;
-    }
-
-    const stack = standardizeStackTrace(stacktrace, message);
-    if (ignoreMessageOrException(message, stack)) {
-      res.sendStatus(statusCodes.BAD_REQUEST);
-      return null;
-    }
-    if (!debug) {
-      res.sendStatus(statusCodes.ACCEPTED);
-    }
+  const rtvs = await latestRtv();
+  // Drop requests from RTVs that are no longer being served.
+  if (rtvs.length > 0 && !rtvs.includes(version)) {
+    return res.sendStatus(statusCodes.OK);
+  }
 
-    return unminify(stack, version)
-      .then(stack => {
-        const reqUrl =
-          req.method === 'POST'
-            ? `${req.originalUrl}?${buildQueryString()}`
-            : req.originalUrl;
-        const normalizedMessage = /^[A-Z][a-z]+: /.test(message)
-          ? message
-          : `Error: ${message}`;
-        const event = {
-          serviceContext: {
-            service: logTarget.serviceName,
-            version: logTarget.versionId,
-          },
-          message: [normalizedMessage].concat(stack).join('\n'),
-          context: {
-            httpRequest: {
-              method: req.method,
-              url: reqUrl,
-              userAgent: req.get('User-Agent'),
-              referrer: referrer,
-            },
-          },
-        };
-        const logPromise = logEvent(logTarget.log, event);
+  const event = await buildEvent(req, reportingParams, logTarget);
```

Comment:
Re: `unminify` failing. We should be catching here to keep the same behavior.

Reply:
Done.